build/mac/BUILD.gn

```diff
@@ -393,250 +255,191 @@ if (install_omaha4_with_pkg) {
+      args += [ "--notary-arg=$arg" ]
+    }
+  }
+  if (is_universal_binary) {
+    args += [ "--universal" ]
+    deps += [ ":universalize" ]
+  }
 }
 
-action("sign_dmg") {
-  script = "//build/gn_run_binary.py"
-  shell_script = "sign_dmg.sh"
-
-  args = [
-    rebase_path(shell_script, root_build_dir),
-    rebase_path(unsigned_dmg_path),
-    rebase_path(signed_dmg_path),
-    keychain_db,
-    mac_signing_identifier,
-    "-r=designated => identifier \"$chrome_product_full_name\" and certificate leaf = H\"$mac_signing_identifier\"",
+copy("signing_config") {
+  visibility = [ ":package" ]
+  sources = [
+    "signing/internal_config.py",
+    "signing/internal_invoker.py",
   ]
-
-  inputs = [
-    script,
-    shell_script,
-    unsigned_dmg_path,
-  ]
-
-  outputs = [ "${root_out_dir}/$mac_signed_output_prefix/$brave_dmg" ]
-
-  deps = [ ":create_dmg" ]
+  outputs = [ "$packaging_dir/signing/{{source_file_part}}" ]
 }
 
-group("finalize_dmg_pkg") {
-  deps = []
-  if (skip_signing) {
-    deps += [ ":create_dmg" ]
-  } else {
-    if (mac_installer_signing_identifier == "") {
-      deps += [ ":sign_app" ]
-    } else {
-      deps += [ ":copy_dmg_pkg" ]
-      if (sparkle_dsa_private_key_file != "") {
-        deps += [ ":generate_dsa_sig_for_dmg" ]
-      }
+copy("browser_install_script") {
+  deps = [ "//chrome/updater/mac:browser_install_script" ]
+  sources = [ "$root_out_dir/$browser_product_name Packaging/updater/.install" ]
+  outputs = [ "$packaging_dir/keystone_install.sh" ]
+}
 
-      if (sparkle_eddsa_private_key != "") {
-        deps += [ ":generate_eddsa_sig_for_dmg" ]
-      }
+copy("dmg_background_image") {
+  sources = [ "dmg-background.png" ]
+  outputs = [ "$packaging_dir/chrome_dmg_background.png" ]
+}
 
-      if (build_delta_installer && last_chrome_installer != "") {
-        deps += [ ":build_delta_installer" ]
-        if (sparkle_eddsa_private_key != "") {
-          deps += [ ":generate_eddsa_sig_for_delta" ]
-        }
-      }
-    }
+copy("dmg_icon") {
+  dmg_icon_file_name = "dmg"
+  if (!is_release_channel) {
+    dmg_icon_file_name = "dmg-$brave_channel"
   }
+  sources = [ "$dmg_icon_file_name.icns" ]
+  outputs = [ "$packaging_dir/chrome_dmg_icon.icns" ]
 }
 
-copy("copy_notarize_script") {
-  sources = [ "notarize_dmg_pkg.py" ]
+copy("ds_store") {
+  ds_store_file_name = "DS_Store"
+  if (!is_release_channel) {
+    ds_store_file_name = "DS_Store.$brave_channel"
+  }
+  sources = [ "DS_Store-files/$ds_store_file_name" ]
+  outputs = [ "$packaging_dir/chrome_dmg_dsstore" ]
+}
 
+copy("helper_entitlements") {
+  visibility = [ ":package" ]
+  sources = [ "//brave/app/helper-entitlements.plist" ]
+  public_deps = [ "//chrome/installer/mac" ]
   outputs = [ "$packaging_dir/{{source_file_part}}" ]
 }
 
-action("notarize_dmg_pkg") {
-  script = "$packaging_dir/notarize_dmg_pkg.py"
-
-  if (is_official_build) {
-    development_arg = ""
+if (is_official_build) {
+  copy("provisionprofile") {
```

Comment:
I'd rename all these `copy()` targets to have `copy_` prefix in the name to distinguish between file-only deps and real actions.

Reply:
Done in 1d50cabb8a273af8e49aaa8c6bfc44c7eb8e0fb2.